pkg/kratos/service.go

```diff
@@ -178,17 +206,24 @@ func (s *Service) CreateBrowserLoginFlow(
 		Refresh(refresh).
 		Cookie(httpHelpers.CookiesToString(cookies))
 
-	if loginChallenge != "" {
-		request = request.LoginChallenge(loginChallenge)
-	} else if loginChallenge == "" && returnTo == "" {
-		return nil, nil, fmt.Errorf("no return_to or login_challenge was provided")
+	if !s.oidcWebAuthnSequencingEnabled {
+		if loginChallenge != "" {
+			request = request.LoginChallenge(loginChallenge)
+		} else if loginChallenge == "" && returnTo == "" {
+			return nil, nil, fmt.Errorf("no return_to or login_challenge was provided")
+		}
 	}
 
 	flow, resp, err := request.Execute()
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// Populate the flow with the hydra login req, so that the UI can retrieve this info
+	if s.oidcWebAuthnSequencingEnabled {
+		s.hydrateKratosLoginFlow(ctx, flow)
+	}
```

Comment:
having this here after the clause has been evlauted in https://github.com/canonical/identity-platform-login-ui/pull/465/files#diff-5e4ba31969008b305d67765d5d4c9bd50d9bcc49571f1594e97d2b152ddf1378R209 seems a but strange

is there anything we can reuse from the request instead of this?would that be cleaner ?

Reply:
I tried to refactor it, should be better now.